elyra/pipeline/processor_local.py

```diff
@@ -0,0 +1,149 @@
+    """
+    _type = 'local'
+
+    @property
+    def type(self):
+        return self._type
+
+    def process(self, pipeline):
+        """
+        Process a pipeline locally.
+        The pipeline execution consists on properly ordering the operations
+        based on it's dependency graph and than delegating the execution
+        to proper executor (e.g. papermill to notebooks)
+        """
+        timestamp = datetime.now().strftime('%m%d%H%M%S')
+        pipeline_name = f'{pipeline.name}-{timestamp}'
+
+        pipeline_work_dir = LocalPipelineProcessor._get_pipeline_work_dir(pipeline_name)
+        self.log.info(f'Processing Pipeline : {pipeline.name} at {pipeline_work_dir}')
+
+        # Sort operations based on dependency graph (topological order)
+        operations = LocalPipelineProcessor._sort_operations(pipeline.operations)
+        for operation in operations:
+            absolute_filename = self.get_absolute_path(operation.filename)
+            self._execute_file(pipeline_work_dir, absolute_filename)
+
+        return PipelineProcessorResponse('', '', object_storage_path=pipeline_work_dir)
+
+    def export(self, pipeline, pipeline_export_format, pipeline_export_path, overwrite):
+        raise NotImplementedError('Local pipelines does not support export functionality')
+
+    def _execute_file(self, work_dir, filepath):
+        if not os.path.isfile(filepath):
+            raise FileNotFoundError(f'Could not find {filepath}')
+
+        notebook_dir = os.path.dirname(filepath)
+        notebook_name = os.path.basename(filepath)
+        stdout = os.path.join(work_dir, f'{notebook_name}.out')
+        stderr = os.path.join(work_dir, f'{notebook_name}.err')
+
+        self.log.debug(f'Processing: {filepath}')
+        with open(stdout, 'w') as stdout_file, open(stderr, 'w') as stderr_file:
+            t0 = time.time()
+            try:
+                # Try executing with configured python kernel
+                papermill.execute_notebook(
+                    input_path=filepath,
+                    output_path=filepath,
+                    stdout_file=stdout_file,
+                    stderr_file=stderr_file,
+                    cwd=notebook_dir
+                )
+            except KeyError:
+                try:
+                    # force default python kernel
+                    papermill.execute_notebook(
+                        input_path=filepath,
+                        output_path=filepath,
+                        stdout_file=stdout_file,
+                        stderr_file=stderr_file,
+                        cwd=notebook_dir,
+                        kernel_name="python3"
+                    )
+                except Exception as ex:
+                    self.log.error(f'Internal error executing {filepath}')
+                    raise RuntimeError(f'Internal error executing {filepath}.'
```

Comment:
Please distinguish this exception message from the outer one. 
```suggestion
                    self.log.error(f'Retry execution of {filepath} failed.')
                    raise RuntimeError(f'Retry execution of {filepath} failed.'
```